jamal-engine/src/main/java/javax0/jamal/engine/Processor.java

```diff
@@ -259,7 +259,9 @@ String getNextMacroBody(final Input input) throws BadSyntaxAt {
                 counter--; // count the closing
                 if (counter == 0) {
                     skip(input, macros.close());
-                    return output.toString();
+                    //return output.toString();
```

Comment:
Good catch, but instead of that I would leave the return here and transform the loop to an infinite loop.